### DISCUSSION TITLE
Overriding quarkus properties at runtime

### DISCUSSION THREAD

**Discussion by talbert83** [keycloak/keycloak#25332] [Q&A] [answered]
Is there a way to override the values defined in `quarkus.properties` file using env variables - as described in the quarkus documentation? I tried it but it was happily ignored.

To be more precise:
Say I want to have log rolling on boot turned off by default and enable it on certain environments only. The quarkus property to use is: `quarkus.log.file.rotation.rotate-on-boot`. I set this to false in the `quarkus.properties` file packaged into my customized keycloak and tried overriding it using a system property at startup (`-Dquarkus.log.file.rotation.rotate-on-boot=true`) or an env variable `QUARKUS_LOG_FILE_ROTATION_ROTATE_ON_BOOT`, but neither worked.

How can I do this? Thank you!

**Comment by vmuzikar** [comment]
Unfortunately, this is not supported. Specifying raw Quarkus options should be considered a limited workaround. Please consider opening an RFE for adding a Keycloak option for your use case.

**Comment by shawkins** [comment, marked as answer]
@talbert83 @vmuzikar since the normal heirarcy for quarkus properties is inhibited, the workaround is to use env references in the quarkus.properties:

```
quarkus.log.file.rotation.rotate-on-boot=${ROTATE_ON_BOOT:false}
```

You should then be able to override via:

```
export ROTATE_ON_BOOT=true
```

**Reply by talbert83** [reply]
That worked indeed. Thanks for your help!
